lnwallet/btcwallet/psbt.go

```diff
@@ -521,3 +521,39 @@ func (b *BtcWallet) FinalizePsbt(packet *psbt.Packet, accountName string) error
 
 	return b.wallet.FinalizePsbt(keyScope, accountNum, packet)
 }
+
+// lookupFirstCustomAccount returns the first custom account found.
+// In theory, it should be only one custom account for the given
+// name. However, due to a lack of check, users could have created
+// custom accounts with various key scopes. This behaviour has
+// been fixed but, we still need to handle this specific case to
+// avoid non-deterministic behaviour implied by 'LookupAccount'.
+func (b *BtcWallet) lookupFirstCustomAccount(
+	name string) (waddrmgr.KeyScope, uint32, error) {
+
+	var (
+		account  *waddrmgr.AccountProperties
+		keyScope waddrmgr.KeyScope
+	)
+	for _, scope := range waddrmgr.DefaultKeyScopes {
```

Comment:
Here the order is not important and we take the first one we find (according to `DefaultKeyScopes`)

If we want to have a "priority" in the search (search first for one key application field rather than another), I will create an array of key scope taking these priorities into account.

Reply:
I guess the order itself isn't important, but that the order is stable to avoid the non-deterministic behavior?

Reply:
Yes it is stable (it's an array:

```
DefaultKeyScopes = []KeyScope{
		KeyScopeBIP0049Plus,
		KeyScopeBIP0084,
		KeyScopeBIP0086,
		KeyScopeBIP0044,
	}
```